### PR TITLE
Avoid generating internal static classes with constant names in Options Source gen

### PR DESCRIPTION
**Pull request by tarekgh**
The Option Source Generator generates static classes with constant names such as `__OptionValidationStaticInstances.__Attributes` and `__OptionValidationStaticInstances.__Validators`. These classes have an `internal` access modifier, which works well when compiling independent projects. However, issues arise when compiling two libraries that both use the Option Source Generator, and one of the libraries is marked with the `InternalsVisibleToAttribute` to grant access to the internals of the first library.

In this scenario, a compilation error occurs because the generated static classes exist in two places and are accessible to one of these libraries. This leads to ambiguous references and compiler errors.

To resolve this issue, a fix has been implemented. When compiling with Language version 11 or higher, the `file` access modifier is used, limiting access to only the file where this class is defined. For earlier language versions, a random suffix is generated and added to the class names.

### CODE REVIEW

**Comment by tarekgh** [comment]
@geeknoid @xakep139 @eiriktsarpalis @layomia

**Review by geeknoid — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.Options/gen/Emitter.cs [hunk 1]
```diff
@@ -13,17 +15,37 @@ namespace Microsoft.Extensions.Options.Generators
...
+        public Emitter(Compilation compilation, bool emitPreamble = true) : base(emitPreamble)
+        {
+            if (((CSharpCompilation)compilation).LanguageVersion >= Microsoft.CodeAnalysis.CSharp.LanguageVersion.CSharp11)
+            {
+                _modifier = "file";
+            }
+            else
+            {
+                _modifier = "internal";
+                string suffix = $"_{new Random().Next():X8}";
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Options/gen/Emitter.cs:40]: Is this the code that prompted https://github.com/dotnet/roslyn-analyzers/pull/6882/? Should we change this code to not use Random, and instead be deterministic?

cc @jaredpar

Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Options/gen/Emitter.cs:40]: I'll fix that. Is it acceptable to use GUID in this case? or do you suggest any other solution?

Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Options/gen/Emitter.cs:40]: This is now tracked by https://github.com/dotnet/runtime/issues/90990

**Pull request merged**